pkg/controller/keyvault/ingress_secret_provider_class.go

```diff
@@ -103,28 +98,40 @@ func (i *IngressSecretProviderClassReconciler) Reconcile(ctx context.Context, re
 		},
 	}
 	logger = logger.WithValues("spc", spc.Name)
-	ok, err := i.buildSPC(ing, spc)
-	if err != nil {
-		logger.Info("failed to build secret provider class for ingress, user input invalid. sending warning event")
-		i.events.Eventf(ing, "Warning", "InvalidInput", "error while processing Keyvault reference: %s", err)
-		return result, nil
+
+	// Checking if we manage the ingress. All false cases without an error are assumed that we don't manage it
+	var isManaged bool
+	if isManaged, err = i.ingressManager.IsManaging(ing); err != nil {
+		return result, fmt.Errorf("determining if ingress is managed: %w", err)
 	}
-	if ok {
-		logger.Info("reconciling secret provider class for ingress")
-		err = util.Upsert(ctx, i.client, spc)
-		if err != nil {
-			i.events.Eventf(ing, "Warning", "FailedUpdateOrCreateSPC", "error while creating or updating SecretProviderClass needed to pull Keyvault reference: %s", err)
+
+	if isManaged {
+		var upsertSPC bool
+
+		if upsertSPC, err = buildSPC(ing, spc, i.config); err != nil {
+			if userErr := err.(buildSPCUserError); userErr != nil {
+				logger.Info("failed to build secret provider class for ingress, user input invalid. sending warning event")
+				i.events.Eventf(ing, "Warning", "InvalidInput", "error while processing Keyvault reference: %s", userErr.Error())
```

Comment:
userError could be at a higher level than strictly buildSPC so in the future we should move it.  But we should include the `UserError` function which has more customer facing error responses